graphql_compiler/macros/macro_edge/helpers.py

```diff
@@ -53,6 +60,137 @@ def get_directives_for_ast(ast):
+    new_directives = []
+    for directive in ast.directives:
+        if directive.name.value == TagDirective.name:
+            renamed_tag_directive, made_changes_to_tag = _replace_tag_names_in_tag_directive(
+                name_change_map, directive)
+            made_changes = made_changes or made_changes_to_tag
+            new_directives.append(renamed_tag_directive)
+        elif directive.name.value == FilterDirective.name:
+            filter_with_renamed_args, made_changes_to_filter = (
+                _replace_tag_names_in_filter_directive(name_change_map, directive))
+            made_changes = made_changes or made_changes_to_filter
+            new_directives.append(filter_with_renamed_args)
+        else:
+            new_directives.append(directive)
+    return made_changes, new_directives
+
+
+def replace_tag_names(name_change_map, ast):
+    """Replace tag names that are already in use in the whole AST."""
+    if not isinstance(ast, (Field, InlineFragment, OperationDefinition)):
+        return ast
+
+    made_changes = False
+
+    # Recurse
+    new_selection_set = None
+    if ast.selection_set is not None:
+        new_selections = []
+        for selection_ast in ast.selection_set.selections:
+            new_selection_ast = replace_tag_names(name_change_map, selection_ast)
+
+            if selection_ast is not new_selection_ast:
+                # Since we did not get the exact same object as the input, changes were made.
+                # That means this call will also need to make changes and return a new object.
+                made_changes = True
+
+            new_selections.append(new_selection_ast)
+        new_selection_set = SelectionSet(new_selections)
+
+    # Process the current node
+    made_changes_at_this_node, new_directives = _replace_tag_names_at_current_node(
+        name_change_map, ast)
+    made_changes = made_changes or made_changes_at_this_node
+
+    if not made_changes:
+        # We didn't change anything, return the original input object.
+        return ast
+
+    new_ast = copy(ast)
+    new_ast.selection_set = new_selection_set
+    new_ast.directives = new_directives
+    return new_ast
+
+
+def generate_disambiguations(existing_names, new_names):
+    """Return a dict mapping the new names to similar names not conflicting with existing names.
+    Args:
+        existing_names: set of strings, the names that are already taken
+        new_names: set of strings, the names that might coincide with exisitng names
+    Returns:
+        dict mapping the new names to other unique names not present in existing_names
+    """
+    name_mapping = dict()
+    for name in new_names:
+        # We try adding different suffixes to disambiguate from the existing names.
+        disambiguation = name
+        index = 0
+        while disambiguation in existing_names or disambiguation in name_mapping:
+            disambiguation = disambiguation + '_macro_edge_' + str(index)
```

Comment:
This name resolution seems word. If you fail two times it generates, name+'_macro_edge_1_macro_edge_2'. Are you sure you didn't want simply name+'_macro_edge_2'